src/meta/meta_ckpt.c

```diff
@@ -510,6 +512,9 @@ __wt_meta_ckptlist_get(
         ckpt = &ckptbase[slot];
         ckpt->order = (slot == 0) ? 1 : ckptbase[slot - 1].order + 1;
         __wt_seconds(session, &ckpt->sec);
+        /* Update time value for most recent checkpoint. Don't let it move backwards. */
+        if (ckpt->sec > conn->ckpt_most_recent)
+            conn->ckpt_most_recent = ckpt->sec;
```

Comment:
I think this might be vulnerable and may need an atomic cas loop. This code path is called (currently only) from `checkpoint_lock_dirty_tree` which does not hold any locks and can be called by multiple threads if they're checkpointing different trees individually. Two threads could race, both pass the conditional and both update `ckpt_most_recent` with different values and could move `ckpt_most_recent` backward after a backup started. While it is unlikely, with the right scheduling and delay, it is possible.

Reply:
Yup.  Good catch.